kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerRegistration.java

```diff
@@ -93,42 +159,42 @@ void forEachMessage(ConsumerRecord<String, byte[]> event, KafkaServerEventProces
         forEach(topic2Message, event, eventProcessor);
     }
 
-    private synchronized <T extends SignalDescBase> void forEach(Map<String, Map<T, Collection<String>>> topic2SignalBase,
+    private synchronized <T extends SignalDescBase> void forEach(Map<String, Map<T, Map<DeploymentId, String>>> topic2SignalBase,
                                                                  ConsumerRecord<String, byte[]> event,
                                                                  KafkaServerEventProcessor<T> processor) {
-        Map<T, Collection<String>> signalInfo = topic2SignalBase.get(event.topic());
+        Map<T, Map<DeploymentId, String>> signalInfo = topic2SignalBase.get(event.topic());
         if (signalInfo != null) {
-            for (Map.Entry<T, Collection<String>> entry : signalInfo.entrySet()) {
+            for (Map.Entry<T, Map<DeploymentId, String>> entry : signalInfo.entrySet()) {
                 T signal = entry.getKey();
-                for (String deploymentId : entry.getValue()) {
+                for (String deploymentId : entry.getValue().values()) {
                     processor.accept(event, deploymentId, signal);
                 }
             }
         }
     }
 
-    private <T extends SignalDescBase> void addTopics(Map<String, Map<T, Collection<String>>> topic2SignalBase,
-                                                      String deploymentId,
+    private <T extends SignalDescBase> void addTopics(Map<String, Map<T, Map<DeploymentId, String>>> topic2SignalBase,
+                                                      DeploymentId deploymentId,
                                                       Collection<T> signals) {
         for (T signal : signals) {
             if (!signal.getIncomingNodes().isEmpty()) {
                 topic2SignalBase.computeIfAbsent(topicFromSignal(signal), k -> new HashMap<>()).computeIfAbsent(
-                        signal, k -> new ArrayList<>()).add(deploymentId);
+                                                                                                                signal, k -> new LinkedHashMap<>()).put(deploymentId, deploymentId.toString());
```

Comment:
this map will cause a race condition as the last to be deployed it will the one getting all the attention

Reply:
thats the purpose, to start process in the latest deployed one.